types/ramda/index.d.ts

```diff
@@ -1575,6 +1577,12 @@ declare namespace R {
         multiply(a: number, b: number): number;
         multiply(a: number): (b: number) => number;
 
+        /**
+         * Moves an item, at index `from`, to index `to`, in a list of elements.
+         * A new list will be created containing the new elements order.
+         */
+        move: CurriedFunction3<number, number, any[], any[]>;
```

Comment:
The `any[]` as an input and output type is suboptimal, since it strips away the array element type from the resulting array, even though the function guarantees that it will be the same as in the input array.
Also, instead of an array as an input type, the `ReadonlyArray` type should be used, since the input value is not modified. This way a larger number of valid inputs are accepted.

So the types should be `ReadonlyArray<T>` and `T[]`, for any `T`. Since you can't add a type parameter to a type alias like `CurriedFunction3` the definition needs to be expanded like with the other functions in the typings

Reply:
Thanks @googol for the review. As per your suggestions, I have used types `ReadonlyArray<T> `and `T[]` in the type definitions and also have expanded the definitions instead of using `CurriedFunction3`.

The reason I chose CurriedFunction3 at first is I wanted to cover all the permutations of the curried funtions. e.g. The move function accepts 3 arguments viz. from, to and list. Possible Combinations are:

1) (from, to, list) => list
2) (from, to) => (list) => list
3) (from) => (to, list) => list
4) (from) => (to) => (list) => list

The types that I've written cover the first three possibilities. However, if I add type definition for 4th possibility, the typescript compiler gets confused between 3rd and 4th definition and gives error.

Type definition for Possibility 3:
`move(from: number): (<T>(to: number, list: ReadonlyArray<T>) => T[]);`

Type definition for Possibility 4:
`move(from: number): (to: number) => <T>(list: ReadonlyArray<T>) => T[];`

Below test fails if we have both the type definitions 3 & 4 present.
```
const moveCurried3 = R.move(0);
const moveCurried4 = moveCurried3(2);
moveCurried4<string>(sampleList); // => ['b', 'c', 'a', 'd', 'e', 'f']
```

Could you please advise how can we incorporate types for all the 4 possible scenarios ?

Reply:
Hmm yeah that's an interesting problem. There are some examples of how to do it in the typings, but it's not used on every function. The idea is to return an interface from the `move(from: number) =>` case, that includes two call signatures. For an example of this, see the `propEq` function typings. Note that for the last case, where each parameter is passed individually, the type parameter can be left to just the last parameter.

Reply:
@googol  Your suggestion (The idea is to return an interface from the `move(from: number) =>` case, that includes two call signatures.) is just brilliant! Works like a charm. Thanks for the advice 👍 
Have made changes accordingly. Now, we have typings for R.move that cover all 4 scenarios.